### PR TITLE
Updated versions of plugins and replaced deprecated methods and classes

### PR DESCRIPTION
**Pull request by andreeapad**
The version of Mockito that seems to work with the lastest version of Powermock(1.7.4) is 2.8.8, and the latest one in Maven Central is 2.8.47([https://github.com/powermock/powermock/wiki/Mockito](url)).
Since org.mockito.runners.MockitoJUnitRunner is deprecated, I replaced it with org.mockito.junit.MockitoJUnitRunner, the same with some other methods.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <parent>
     <groupId>com.hotels</groupId>
     <artifactId>hotels-oss-parent</artifactId>
-    <version>2.0.6</version>
+    <version>2.2.0</version>
```
Comment (massdosage) [inline pom.xml]: Please change this to 2.3.0 so we can also test out the changes just released the parent around releasing artifacts to Sonatype.

File: waggle-dance-api/src/test/java/com/hotels/bdp/waggledance/api/validation/validator/TunnelRouteValidatorTest.java [hunk 2]
```diff
@@ -27,7 +27,6 @@
 import org.mockito.runners.MockitoJUnitRunner;
 
 import com.hotels.bdp.waggledance.api.validation.constraint.TunnelRoute;
-import com.hotels.bdp.waggledance.api.validation.validator.TunnelRouteValidator;
```
Comment (massdosage) [inline waggle-dance-api/src/test/java/com/hotels/bdp/waggledance/api/validation/validator/TunnelRouteValidatorTest.java]: Unused import - well spotted! Extra gold star for removing it ;)

**Comment by massdosage** [comment]
The Travis build is failing like so:

`[ERROR] Failed to execute goal on project waggle-dance-core: Could not resolve dependencies for project com.hotels:waggle-dance-core:jar:2.3.8-SNAPSHOT: Could not find artifact org.mockito:mockito-core:jar:2.8.8 in central (http://repo.maven.apache.org/maven2) -> [Help 1]
`
Are you sure your changes build?

**Comment by massdosage** [comment]
Also, please update the version in all the poms to 2.4.0-SNAPSHOT as this feels like a minor version change since we're updating a few dependencies.

File: pom.xml [hunk 3]
```diff
@@ -32,17 +32,17 @@
...
-    <cobertura.version>2.0.3</cobertura.version>
-    <cobertura.maven.plugin.version>2.6</cobertura.maven.plugin.version>
+    <cobertura.version>2.1.1</cobertura.version>
+    <cobertura.maven.plugin.version>2.7</cobertura.maven.plugin.version>
     <maven.release.plugin.version>2.5.3</maven.release.plugin.version>
     <spring-platform.version>Brussels-SR5</spring-platform.version>
     <hadoop.version>2.7.2</hadoop.version>
-    <hive.version>2.3.0</hive.version>
-    <dropwizard.version>1.0.5</dropwizard.version>
+    <hive.version>2.3.3</hive.version>
```
Comment (massdosage) [inline pom.xml]: Call this one out in the changelog as it does affect usage outside of the internals of this project - say something like "Upgraded default Hive version from 2.3.0 to 2.3.3"

File: waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/MetaStoreMapping.java [hunk 4]
```diff
@@ -51,14 +51,14 @@
...
    *          have been stripped already)
-   * @return throws {@link NotAllowedException} when the metastore mapped by this class does not have the correct
-   *         permission otherwise calls {{@link #getClient().create_database(database)}
-   * @throws TException
-   * @throws MetaException
-   * @throws InvalidObjectException
-   * @throws AlreadyExistsException
+   * @throws NotAllowedException when the metastore mapped by this class does not have the correct permission otherwise
+   *           calls getClient().create_database(database)
+   * @throws TException Thrift exception
```
Comment (massdosage) [inline waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/MetaStoreMapping.java]: Usually the description is what can lead to the exception being thrown, like you've done on line 61. It's not that straightforward to actually find the reasons as Hive's own documentation doesn't include them :( so I'm just going to put what I think the best we can do here is. For this one - "if a Thrift error occurs".

File: waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/MetaStoreMapping.java [hunk 5]
```diff
@@ -51,14 +51,14 @@
...
+   * @throws MetaException Hive exception
```
Comment (massdosage) [inline waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/MetaStoreMapping.java]: if a problem occurs interacting with the Hive MetaStore

File: waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/MetaStoreMapping.java [hunk 6]
```diff
@@ -51,14 +51,14 @@
...
+   * @throws InvalidObjectException Hive exception
```
Comment (massdosage) [inline waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/MetaStoreMapping.java]: if Hive considers the database invalid

File: waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/PanopticOperationHandler.java [hunk 7]
```diff
@@ -27,21 +27,34 @@
...
+   *
+   * @return list of all databases
    */
   List<String> getAllDatabases();
 
   /**
    * Implements {@link HMSHandler#get_database(String)} over multiple metastores
+   *
+   * @param databasePattern pattern
+   * @return list of all databases with a pattern
```
Comment (massdosage) [inline waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/PanopticOperationHandler.java]: list of all databases that match the passed pattern

File: waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/PanopticOperationHandler.java [hunk 8]
```diff
@@ -27,21 +27,34 @@
...
    * Implements {@link HMSHandler#get_all_databases()} over multiple metastores
+   *
+   * @return list of all databases
    */
   List<String> getAllDatabases();
 
   /**
    * Implements {@link HMSHandler#get_database(String)} over multiple metastores
+   *
+   * @param databasePattern pattern
```
Comment (massdosage) [inline waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/PanopticOperationHandler.java]: pattern to match

File: waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/PanopticOperationHandler.java [hunk 9]
```diff
@@ -27,21 +27,34 @@
...
+   * @return list of all databases with a pattern
    */
   List<String> getAllDatabases(String databasePattern);
 
   /**
    * Implements {@link HMSHandler#get_table_meta(String, String, List)} over multiple metastores
+   *
+   * @param databasePatterns database patterns
```
Comment (massdosage) [inline waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/PanopticOperationHandler.java]: database patterns to match

File: waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/PanopticOperationHandler.java [hunk 10]
```diff
@@ -27,21 +27,34 @@
...
+   * @param tablePatterns table patterns
```
Comment (massdosage) [inline waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/PanopticOperationHandler.java]: table patterns to match

File: waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/PanopticOperationHandler.java [hunk 11]
```diff
@@ -27,21 +27,34 @@
...
+   * @return list of table metadata
```
Comment (massdosage) [inline waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/PanopticOperationHandler.java]: you're missing `@param tableTypes table types to match`

File: pom.xml [hunk 12]
```diff
@@ -32,17 +32,17 @@
   </modules>
 
   <properties>
-    <cobertura.version>2.0.3</cobertura.version>
-    <cobertura.maven.plugin.version>2.6</cobertura.maven.plugin.version>
+    <cobertura.version>2.1.1</cobertura.version>
+    <cobertura.maven.plugin.version>2.7</cobertura.maven.plugin.version>
     <maven.release.plugin.version>2.5.3</maven.release.plugin.version>
     <spring-platform.version>Brussels-SR5</spring-platform.version>
```
Comment (courtsVII) [inline pom.xml]: There is a new Spring BOM release `Cairo-RELEASE` shall we move to this @massdosage ?

Comment (massdosage) [reply pom.xml]: I'd like to do that but I'd prefer holding off for this PR as I don't want to overcomplicate things and I worry that might introduce bigger changes. Also, spring platform as a "thing" is end-of-life and the recommendation from them is to replace this with the `spring-boot-dependencies` bom. I'll create a separate ticket for that in a bit.

**Pull request merged**